src/corelibs/U2Core/src/tasks/AddSequencesToAlignmentTask.cpp

```diff
@@ -186,38 +186,47 @@ void AddSequencesFromFilesToAlignmentTask::prepare() {
     AddSequenceObjectsToAlignmentTask::prepare();
     for (const QString &fileWithSequencesUrl : qAsConst(urlList)) {
         QList<FormatDetectionResult> detectedFormats = DocumentUtils::detectFormat(fileWithSequencesUrl);
-        if (!detectedFormats.isEmpty()) {
-            IOAdapterFactory *factory = AppContext::getIOAdapterRegistry()->getIOAdapterFactoryById(BaseIOAdapters::LOCAL_FILE);
-            DocumentFormat *format = detectedFormats.first().format;
-            loadTask = new LoadDocumentTask(format->getFormatId(), fileWithSequencesUrl, factory);
-            addSubTask(loadTask);
-        } else {
-            setError("Unknown format");
+        if (detectedFormats.isEmpty()) {
+            setError(tr("Unsupported document format: %1").arg(fileWithSequencesUrl));
+            continue;
         }
+        IOAdapterFactory *factory = AppContext::getIOAdapterRegistry()->getIOAdapterFactoryById(BaseIOAdapters::LOCAL_FILE);
+        DocumentFormat *format = detectedFormats.first().format;
+        loadTask = new LoadDocumentTask(format->getFormatId(), fileWithSequencesUrl, factory);
+        addSubTask(loadTask);
     }
 }
 
 QList<Task *> AddSequencesFromFilesToAlignmentTask::onSubTaskFinished(Task *subTask) {
-    const QList<Task *> emptySubTasks;  // Helper constant. This method never returns any subtasks.
-
     propagateSubtaskError();
-    if (isCanceled() || hasError()) {
-        return emptySubTasks;
-    }
+    CHECK(!isCanceled() && !hasError(), {});
```

Comment:
Could be even better - `CHECK_OP(stateInfo, {});`

Reply:
Fixed